### ISSUE TITLE
MetadataValue returning a ValueError

### ISSUE THREAD

**Issue by JustGui** [dagster-io/dagster#16557]
### Dagster version

1.4.12

### What's the issue?

I'm returning **metadata** using **Output** on an @asset for a while but I'm starting to have many errors when returning a preview with this method: "preview": `MetadataValue.md(df.head().to_markdown())` [(proposed in your tutorial)](https://docs.dagster.io/tutorial/saving-your-data)

The error:
`ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`

If I do a `get_dagster_logger().info(df.head())` I don't have this problem; seems like **MetadataValue.md** is doing an operation, not done by **get_dagster_logger()**, that may cause a panda error with some specific dataframes

### How to reproduce?

It's not with all my dataframes and I'm struggling to find which piece of data in my df could return this issue

### Deployment type

Other Docker-based deployment

### Deployment details

Dagster and the python code are both in a docker
Image of the docker: `python:3.9-bullseye`

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by tacastillo** [comment]
Thanks for raising! As mentioned on Slack, I can't reproduce this, so I'll use this issue to track if other people are hitting and then figure out why based on the reports.

**Comment by JustGui** [comment]
As explained on slack, we found the cause of the issue.


**TL;DR:** You shouldn't use `MetadataValue.md(df.head().to_markdown())` if you retrieve data from a BigQuery table with a column in RECORD format as the df is malstructured and raise an error when using `.to_markdown()`


**Complete explanation:**
1. First we get data from BigQuery using Dagster's BigQueryResource

   The BigQuery table has two columns:
           a. `column1`: **record** type with two fields: field1 and field2
           b. `column2`: **string**
           
    Column1 would give in a JSON format: [{'field1': '1','field2": '0x123'}, {'field1': '2','field2': '0x1234'}]
    
2. Then if we look at the dataframe created we realize that the comma separating `{'field1': '1','field2": '0x123'}` and `{'field1': '2','field2': '0x1234'}` **do not exists**, the JSON is malstructured

3. If we repush this df to BigQuery we have no error so BigQuery understands this JSON without any comma separator

Unfortunatly,  this JSON structure in **df["column1"]** raises an error when using `.to_markdown`, which is mandatory for `MetadataValue.md`
